lib/Transforms/IPO/PassManagerBuilder.cpp

```diff
@@ -899,6 +903,13 @@ void PassManagerBuilder::populateLTOPassManager(legacy::PassManagerBase &PM) {
 
   if (VerifyOutput)
     PM.add(createVerifierPass());
+
+  // After all optimizations have been performed, struct field cache analysis will be performed if enabled
```

Comment:
This is not necessarily the best place to have this pass. For instance some transforms may make your analysis harder to do, so it may make sense to have it early. Of course, this depends on what goes into this pass, so this is premature to figure out where exactly this goes. May be add a FIXME saying something like we need to figure out the right place for this pass.